src/utils/provideLocalStorageDataClass.ts

```diff
@@ -40,12 +40,24 @@ export function provideLocalStorageDataClass(
             ? items
             : items.filter((item) =>
                 Object.entries(filters).every(
-                  ([filterAttribute, filterValue]) =>
-                    item[filterAttribute] === filterValue ||
-                    (filterValue === 'true' &&
-                      item[filterAttribute] === true) ||
-                    (filterValue === 'false' &&
-                      item[filterAttribute] === false),
+                  ([filterAttribute, { value: filterValue, opCode }]) => {
+                    const itemValue = item[filterAttribute]
+
+                    if (opCode === 'neq') {
+                      return (
+                        (typeof itemValue !== 'boolean' &&
+                          itemValue !== filterValue) ||
+                        (filterValue === 'true' && itemValue !== true) ||
+                        (filterValue === 'false' && itemValue !== false)
+                      )
```

Comment:
* [ ] This should evaluate to `true` if, e.g., `filterValue === 'foo'` and `itemValue` is boolean.

Finding this out by reading the code took about 5 minutes.
To escape boolean hell, how about reusing the "eq" code and negating the overall result?